x-pack/plugins/ml/public/application/services/state_service.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Subject } from 'rxjs';
+
+export class StateService {
```

Comment:
looks like this class can be abstract with `init` declared as a protected abstract method?

Reply:
it has a `destroy` method that relies on the `unsubscribeAll` subject, so I guess an abstract class isn't going to work here